src/ObjectUtils/LinuxMap.cpp

```diff
@@ -113,30 +102,16 @@ ErrorMessageOr<std::vector<ModuleInfo>> ParseMaps(std::string_view proc_maps_dat
     uint64_t end = std::stoull(addresses[1], nullptr, 16);
     bool is_executable = tokens[1].size() == 4 && tokens[1][2] == 'x';
 
-    auto iter = address_map.find(module_path);
-    if (iter == address_map.end()) {
-      address_map[module_path] = {start, end, is_executable};
-    } else {
-      AddressRange& address_range = iter->second;
-      address_range.start_address = std::min(address_range.start_address, start);
-      address_range.end_address = std::max(address_range.end_address, end);
-      address_range.is_executable |= is_executable;
-    }
-  }
-
-  std::vector<ModuleInfo> result;
-  for (const auto& [module_path, address_range] : address_map) {
-    // Filter out entries which are not executable
-    if (!address_range.is_executable) continue;
+    // Skip non-executable mappings
+    if (!is_executable) continue;
```

Comment:
This should do the fix for the case documented in b/191732187. I wonder if there are real world cases with multiple executable mappings per module. Probably there is little use in asking that - we'll find out ...

Reply:
yep - this is currently assumes that there is only one executable segment. I think for now it is good enough.